src/custom/utils/trade.ts

```diff
@@ -88,14 +89,36 @@ export async function postOrder(params: PostOrderParams): Promise<string> {
     partiallyFillable: false // Always fill or kill
   }
 
-  const signature = (await signOrder({
+  let signature: Signature | null = null
+  let signingScheme = DEFAULT_SIGNING_SCHEME
+
+  const signatureParams: SignOrderParams = {
     chainId,
     signer,
     order: unsignedOrder,
-    signingScheme: SIGNING_SCHEME
-  })) as EcdsaSignature // Only ECDSA signing supported for now
+    signingScheme
+  }
+
+  try {
+    signature = (await signOrder(signatureParams)) as EcdsaSignature // Only ECDSA signing supported for now
+  } catch (e) {
+    if (e.code === METAMASK_SIGNATURE_ERROR) {
+      // We tried to sign order the nice way.
+      // That works fine for regular MM addresses. Does not work for Hardware wallets, though.
+      // See https://github.com/MetaMask/metamask-extension/issues/10240#issuecomment-812672324
+      // So, when that specific error occurs, we know this is a problem with MM + HW.
+      // Then, we fallback to ETHSIGN.
+      console.info('[trade:postOrder] Wallet does not support EIP712. Trying ETHSIGN instead', e)
+      signingScheme = SigningScheme.ETHSIGN
+      signature = (await signOrder({ ...signatureParams, signingScheme })) as EcdsaSignature // Only ECDSA signing supported for now
+    } else {
+      // Some other error signing. Let it bubble up.
+      console.error(e)
+      throw e
+    }
+  }
 
-  const signatureData = signature.data.toString()
+  const signatureData = signature?.data.toString()
```

Comment:
are we assuming that i the `signatureData` here is `undefined` that it was because there was an error and it already `throw`-d the error above? otherwise i would add a better check for this since I'm not sure posting an order with `undefined` `signatureData` is bueno

Reply:
Makes sense, will revisit